### PR TITLE
[Release Tooling] Validation workaround for Xcode 15.3b3

### PR DESCRIPTION
**Pull request by ncooke3**
context in b/327020913

Validated locally by building one xcframework. It's possible this could blow up when building all of the xcframeworks, but hopefully not.

### CODE REVIEW

File: ReleaseTooling/Sources/ZipBuilder/FrameworkBuilder.swift
Comment (paulb777) [inline ReleaseTooling/Sources/ZipBuilder/FrameworkBuilder.swift:674]: Please add a comment with a reference to the bug so future us can find the context.

**Review by paulb777 — APPROVED** [review]
LGTM after comment, green CI and confirm that we don't do something strange on catalyst and macos like writing a one-line plist file.

File: ReleaseTooling/Sources/ZipBuilder/FrameworkBuilder.swift
Comment (ncooke3) [reply ReleaseTooling/Sources/ZipBuilder/FrameworkBuilder.swift:674]: > ... and confirm we don't do something strange on catalyst and macos like writing a one-line plist file.

Now that you mention it, I'm noticing that no plist is generated for macOS/Catalyst. That looks to be expected, correct?

Comment (paulb777) [reply ReleaseTooling/Sources/ZipBuilder/FrameworkBuilder.swift:674]: Correct

File: ReleaseTooling/Sources/ZipBuilder/FrameworkBuilder.swift
Comment (ncooke3) [inline ReleaseTooling/Sources/ZipBuilder/FrameworkBuilder.swift:683]: I may want to make this `.xml` in the future. The binary format is harder to diff from the command line. I'll change it if there is more plist related work down the road.

**Comment by guppy-jpf** [comment]
@ncooke3 

sorry to bother you if you're busy, but quick question:

you say above:
> context in b/327020913

and also in the code:
> // The minimum OS version is set to 100.0 to work around b/327020913.

i'm just curious what "b/327020913" is ... is that some internal google bug tracker reference?

the reason i ask is, i'm in the process of trying to work through a similar issue with a 3rd-party library vendor (best available ref [here](https://github.com/AppsFlyerSDK/AppsFlyerFramework/issues/263#issuecomment-1992764197); sorry, the actual vendor doesn't track issues on github), and their contention is that the issue is caused by an apple bug ...  so i'm curious if you guys also consider this to be an apple bug that you're trying to work around, or do you feel it's just stricter (but not erroneous) checking on the part of xcode 15.3?

if you have time for a quick reply, i'd appreciate it.
thanks!

**Comment by paulb777** [comment]
Yes. The "b" reference is an internal bug tracker and yes we consider it an Apple bug:  Apple should allow SDKs to support a lower minimum OS version than the app and it's very strange that setting the version higher than any available OS is a workaround.

**Comment by paulb777** [comment]
We have an open Apple Feedback about the issue and still waiting for a response.

**Comment by paulb777** [comment]
Feedback (Radar) ID: FB13657402

**Pull request merged**
